### PR TITLE
Add xfailing jit tests to many anchor examples

### PR DESCRIPTION
**Pull request by fritzo**
Addresses #1063

This enables a `--jit` option in many of our examples. None of the examples currently work with `--jit`. The purpose of this PR is to help motivate and prioritize op support in upstream work on `torch.jit`.

This PR is already incorporated into the pytorch-0.4.1 branch, and I'd like to avoid that branch diverging too far from dev.

### CODE REVIEW

File: tests/test_examples.py [hunk 1]
```diff
@@ -90,3 +110,13 @@ def test_cuda(example):
     filename, args = example[0], example[1:]
     filename = os.path.join(EXAMPLES_DIR, filename)
     check_call([sys.executable, filename] + args)
+
+
```
Comment (neerajprad) [inline tests/test_examples.py]: Can we also add a `skipif` on `CI` for now? The examples stage takes quite a bit of travis time.

**Comment by fritzo** [comment]
Thanks for reviewing!

File: tests/test_examples.py [hunk 2]
```diff
@@ -90,3 +110,14 @@ def test_cuda(example):
...
+@pytest.mark.skipif('CI' in os.environ, reason='slow test')
+@pytest.mark.xfail(reason='not jittable')
+@pytest.mark.parametrize('example', JIT_EXAMPLES)
+def test_jit(example):
+    logger.info('Running:\npython examples/{}'.format(example))
+    example = example.split()
+    filename, args = example[0], example[1:]
+    filename = os.path.join(EXAMPLES_DIR, filename)
+    check_call([sys.executable, filename] + args)
```
Comment (jpchen) [inline tests/test_examples.py]: if `CUDA_EXAMPLES` and `JIT_EXAMPLES` are subsets of `CPU_EXAMPLES` what do you think of writing the cuda and jit tests as:
```
@pytest.mark.parametrize('example', CPU_EXAMPLES)
def test_jit(example):
  try:
    ...
    filename, args = example[0], example[1:] + ' --jit'
    ...
  except CalledProcessError as e:
    if 'unrecognized arguments' in e.output:
    # if the examples does not support jit, skip it
      continue
```
or would that run too many needless examples?

Comment (fritzo) [reply tests/test_examples.py]: That seems like too much automation. I'd worry that as we add more logic we'd accidentally disable all tests and silently stop testing (this has happened a couple times already). Therefore I'm more comfortable hand-unrolling the examples lists.

**Comment by jpchen** [comment]
fine by me and fritz

**Pull request merged**